fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -204,6 +204,9 @@
 
     private final Cache<Tuple, FDBRawRecord> preloadCache;
 
+    @Nullable
+    private KeySpacePath keySpacePath;
```

Comment:
I am not clear what this is for. I could see having both a `Subspace` and a `KeySpacePath` and doing away with `SubspaceProvider` in favor of having `FDBRecordStore` figure out how to use the two in various ways. But why `SubspaceProvider` and `KeySpacePath`?